src/components/Club/ClubDepositFunds.tsx

```diff
@@ -0,0 +1,86 @@
+import { useState } from 'react';
+import { Button, Center, Divider, Loader, Modal, Stack, Text } from '@mantine/core';
+import { useDialogContext } from '../Dialog/DialogProvider';
+import { useMutateClub, useQueryClub } from './club.utils';
+import { useBuzz } from '../Buzz/useBuzz';
+import { clubTransactionSchema } from '~/server/schema/buzz.schema';
+import { showSuccessNotification } from '~/utils/notifications';
+import { CurrencyBadge } from '../Currency/CurrencyBadge';
+import { CurrencyIcon } from '../Currency/CurrencyIcon';
+import { z } from 'zod';
+import { Currency } from '@prisma/client';
+import { NumberInputWrapper } from '../../libs/form/components/NumberInputWrapper';
+import { BuzzTransactionButton } from '../Buzz/BuzzTransactionButton';
+
+const schema = clubTransactionSchema.omit({ clubId: true });
+
+export const ClubDepositFunds = ({ clubId }: { clubId: number }) => {
+  const dialog = useDialogContext();
+  const handleClose = dialog.onClose;
+  const { balance: userBalance, balanceLoading: userBalanceLoading } = useBuzz();
+  const { balance, balanceLoading } = useBuzz(clubId, 'Club');
+  const { club, loading } = useQueryClub({ id: clubId });
+  const { depositClubFunds, depositingClubFunds } = useMutateClub();
+  const isLoading = loading || balanceLoading || userBalanceLoading;
+  const [amount, setAmount] = useState(5000);
+
+  const handleSubmit = async () => {
+    await depositClubFunds({ amount, clubId });
+    showSuccessNotification({
+      title: 'Funds have been deposited',
+      message: 'Your funds have been deposited correctly.',
+    });
+    handleClose();
+  };
+
+  return (
+    <Modal {...dialog} title="Deposit your funds into your club" size="sm" withCloseButton>
+      <Stack>
+        <Divider mx="-lg" mb="md" />
+        {isLoading || !club || !balance || !userBalance ? (
```

Comment:
I might be wrong, but seems like the condition should be like the following
```suggestion
        {isLoading && (!club || !balance || !userBalance) ? (
```